### DISCUSSION TITLE
Array Field 在 FormLayout 里面数组的值获取不到？

### DISCUSSION THREAD

**Discussion by pandazki** [alibaba/formily#1531] [Q&A 2.x] [answered]
今天无意中发现，被 FormLayout 包裹的 Array Card 不能获取 InitValue，就做了一下测试，发现只要外层存在 FormLayout，内部所有数组类型的组件似乎赋值和取值都有问题，来请教一下是不是使用姿势有问题。

![image](https://user-images.githubusercontent.com/754493/120287858-39a63480-c2f2-11eb-90c2-3e4390351554.png)

[Demo](https://codesandbox.io/s/silly-leaf-cb4en?file=/App.tsx)

**Comment by janryWang** [comment, marked as answer]
这是个Bug，麻烦提一下issue，下个版本修复

**Reply by pandazki** [reply]
辛苦大佬~ 修好之后刚好学习一下。水平有限，翻了翻源码没有定位到问题。
https://github.com/alibaba/formily/issues/1533

**Comment by liuweiGL** [comment]
应该不是 FormLayout 的问题，而是 ArrayField 外层包裹了 VoidFiled 之后都会无法获取到值。

https://codesandbox.io/s/funny-leaf-zswrz

**Reply by janryWang** [reply]
嗯，跟FormLayout没关系，内核路径计算有问题
